judo-runtime-core-jsl-itest/src/test/java/hu/blackbelt/judo/runtime/core/jsl/util/EnvironmentVariables.java

```diff
@@ -0,0 +1,151 @@
+package hu.blackbelt.judo.runtime.core.jsl.util;
+
+import uk.org.webcompere.systemstubs.SystemStubs;
+import uk.org.webcompere.systemstubs.ThrowingRunnable;
+import uk.org.webcompere.systemstubs.resource.NameValuePairSetter;
+import uk.org.webcompere.systemstubs.resource.SingularTestResource;
+
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static java.util.Collections.emptyMap;
+import static uk.org.webcompere.systemstubs.properties.PropertiesUtils.toStringMap;
+
+/**
+ * A collection of values for environment variables. New values can be
+ * added by {@link #and(String, String)}. The {@code EnvironmentVariables}
+ * object is then used to execute an arbitrary piece of code with these
+ * environment variables being present. Use {@link #execute}
+ * to wrap around test code:
+ * <pre>
+ * &#064;Test
+ * void execute_code_with_environment_variables() throws Exception {
+ *   withEnvironmentVariable("first", "first value")
+ *     .and("second", "second value")
+ *     .and("third", null)
+ *     .execute((){@literal ->} {
+ *       assertEquals("first value", System.getenv("first"));
+ *       assertEquals("second value", System.getenv("second"));
+ *       assertNull(System.getenv("third"));
+ *     });
+ * }
+ * </pre>
+ * @since 1.0.0
+ */
+public class EnvironmentVariables extends SingularTestResource implements NameValuePairSetter<EnvironmentVariables> {
+    protected final Map<String, String> variables;
+
+    /**
+     * Default constructor with an empty set of environment variables. Use {@link #set(String, String)} to
+     * provide some, mutating this, or {@link #and(String, String)} to fork a fresh object with desired
+     * environment variable settings.
+     */
+    public EnvironmentVariables() {
+        this(emptyMap());
+    }
+
+    /**
+     * Construct with an initial set of variables as name value pairs.
+     * @param name first environment variable's name
+     * @param value first environment variable's value
+     * @param others must be of even-numbered length. Name/value pairs of the other values to
+     *               apply to the environment when this object is active
+     */
+    public EnvironmentVariables(String name, String value, String... others) {
+        if (others.length % 2 != 0) {
+            throw new IllegalArgumentException("Must provide even number of parameters");
+        }
```

Comment:
I recommend using [Map.Entry<K, V>](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.Entry.html) instead of `String....`. This way proper key-value arguments can be enforced in compile time

Reply:
Map entry is an interface which is instantiated by the coresponding map type. To use in stream is OK, but it's not recommended to use as instantiated objects.

Reply:
actually, since Java 11 `Map.of(...)` can be use as well. Even simply accepting a Map instance could be sufficient.

Reply:
There is already a constructor with said features. I'm marking this convo as resolved